glib-macros/src/lib.rs

```diff
@@ -518,3 +519,28 @@ pub fn downgrade(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as DeriveInput);
     downgrade_derive::impl_downgrade(input)
 }
+
+/// Derive macro for serializing/deserializing custom structs as [`glib::Variant`]s.
+///
+/// # Example
+///
+/// ```
+/// use glib::prelude::*;
+///
+/// #[derive(glib::Variant, PartialEq, Eq, Debug)]
+/// struct Foo {
+///     some_string: String,
+///     some_int: i32,
+/// }
+///
+/// let v = Foo { some_string: String::from("bar"), some_int: 1 };
+/// let var = v.to_variant();
+/// assert_eq!(var.get::<Foo>(), Some(v));
+/// ```
+///
+/// [`glib::Variant`]: variant/struct.Variant.html
+#[proc_macro_derive(Variant)]
```

Comment:
It's not that great that some macros are prefixed with G and others not :/

Reply:
I agree, but I don't know which option is nicer. Do you have a preference?

Reply:
Let's add the `G` here for consistency with all the other derive macros and create a new issue to discuss this? @bilelmoussaoui

Reply:
Sounds good to me